runtime/vm.go

```diff
@@ -21,6 +21,11 @@ const (
 	MaxFrames = 1024
 )
 
+// internal type used to handle explosion on the stack
+type exploded struct {
```

Comment:
What's the point of this? Seems redundant to me and adds an additional type assertion in `OpArray`, `OpExplode`, and `OpCall`

Reply:
Ah, I see what you're doing. However, there should be a better way of doing this without adding an extra type assertion.

Reply:
Perhaps create an `Exploded` object that extends `Array`. Then use the `Exploded` object directly.

Reply:
`OpExplode` wraps the exploded value on the stack in this type to signal to `OpCall` and `OpArray` to handle the explosion.

In `OpArray` I actually use it to short-circuit having to do a bunch of stack modification (like in `OpCall`), so IMO that's not too bad

In `OpCall`, it is a signal to tell the vm to explode an argument onto the stack. I can't just check for `objects.Explodable` since an explodable object may be passed without actually being exploded.

In order to avoid using this `exploded` type, I'd need to change the calling convention for functions to pass the argument count on the stack, and when I went that route it got complex _really_ fast.

Definitely open to other suggestions though.

Reply:
The `Exploded` type can be an `Object` that holds the contents of `Explode()` within `OpExplode`. That removes the extra type assertion within `OpArray` and `OpCall`

Reply:
good idea, done